lib/anyplayer/selector.rb

```diff
@@ -7,62 +9,93 @@
 #   player = selector.player
 #
 # Needs the PLAYERS constant to contain a list of players.
+module Anyplayer
+  class Selector
+    attr_accessor :verbose
+    attr_reader :errors
 
-class Anyplayer::Selector
-  attr_accessor :verbose
-  attr_reader :errors
+    def initialize
+      @verbose = false
+      @errors = []
+    end
 
-  def initialize
-    @verbose = false
-    @errors = []
-  end
+    # Returns an instance of the first music player that's launched
+    def player
+      PLAYERS.each do |player|
+        player_load(player) or next
 
-  # Returns an instance of the first music player that's launched
-  def player
-    players_for_this_platform.each { |player|
-      player_load(player) or next
-      instance = player_class(player).new
-      return instance if player_launched(instance)
-    }
-    nil
-  end
+        instance = player_class(player).new
+        player_on_platform?(instance) or next
 
+        if player_launched?(instance)
+          return instance
+        end
+      end
+      nil
+    end
 
-  private
 
-  def players_for_this_platform
-    players = Anyplayer::PLAYERS
-    players.reject! { |player| player =~ /_mac$/ } if RUBY_PLATFORM !~ /darwin/
-    players.reject! { |player| player =~ /_windows$/ } if RUBY_PLATFORM !~ /cygwin|mswin|mingw|bccwin|wince|emx/
-    players
-  end
+    private
 
-  def player_load(player)
-    begin
-      require "anyplayer/players/#{player}"
-      $stderr.puts "Loaded #{player}" if verbose
-      true
-    rescue LoadError => e
-      $stderr.puts "Could not load #{player}" if verbose
-      @errors << "Error loading #{player}: #{e.message}"
-      false
+    def platform
+      @platform ||= begin
+        host_os = RbConfig::CONFIG["host_os"]
+        case host_os
+        when /mswin|msys|mingw|cygwin|bccwin|wince|emc/
+          :windows
+        when /darwin|mac os/
+          :mac
+        when /linux/
+          :linux
+        when /solaris|bsd/
+          :unix
+        else
+          @errors << "Unknown operating system #{host_os.inspect}"
+          nil
+        end
+      end
     end
-  end
 
-  def player_class(player)
-    camelized = player.to_s.split(/_/).map{ |word| word.capitalize }.join
-    Anyplayer::const_get(camelized)
-  end
+    def player_load(player)
+      log "#{player}:"
+      begin
+        require "anyplayer/players/#{player}"
+        log "  loaded"
+        true
+      rescue LoadError => e
+        log "  #{e.message}"
+        @errors << "Error loading #{player}: #{e.message}"
+        false
+      end
+    end
+
+    def player_on_platform?(player)
+      result = player.platforms.include?(platform)
+      log "  not on platform" if !result
+      result
+    end
 
-  def player_launched(player)
-    $stderr.puts "#{player.name} launched?" if verbose
+    def player_class(player)
+      camelized = player.to_s.split(/_/).map{ |word| word.capitalize }.join
```

Comment:
Space missing to the left of {.<br>Pass `&:capitalize` as an argument to `map` instead of a block.